src/NewHashMap.java

```diff
@@ -0,0 +1,65 @@
+public class NewHashMap<K, V> {
+	private Node<K, V>[] table;
+	private int sizeOfTable;
+	private static final int TABLE_SIZE = 16;
+	private static final double TABLE_MAX_PERCENT_OCCUPANCY = 0.75;
+
+	public void put(K key, V value) {
+		if (sizeOfTable == 0) {
+			table = new Node[TABLE_SIZE];
+		} else if (table.length * TABLE_MAX_PERCENT_OCCUPANCY <= TABLE_SIZE) {
+			resizeOfTable();
+		}
+		Node<K, V> newNode = new Node<>(key, value);
+		Node<K, V> currentNode = table[hashKeyIndex(key, table.length)];
+		if (currentNode == null) {
+			table[hashKeyIndex(key, table.length)] = newNode;
+		} else if (currentNode.keyValue == key) {
+			currentNode.itemValue = value;
+		} else {
+			while (currentNode.hasNext()) {
```

Comment:
write comment here, i can't understand that

Reply:
I added comments to the project